### PR TITLE
MAM-3675-migrate-profile-header-to-metatext

### PR DESCRIPTION
**Pull request by bnolens**
[MAM-3675 : Migrate Profile Header to MetaText](https://linear.app/theblvd/issue/MAM-3675/migrate-profile-header-to-metatext)

### CODE REVIEW

File: Mammoth/Screens/ProfileScreen/Views/ProfileHeader.swift [hunk 1]
```diff
@@ -709,8 +714,14 @@ final class ProfileField: UIStackView {
             verifiedImage.setContentCompressionResistancePriority(UILayoutPriority(rawValue: 751), for: .horizontal)
         }
         
-        if let description = parseRichText(text: field.value.stripHTML()) {
-            descriptionLabel.attributedText = formatRichText(string: description, label: descriptionLabel, emojis: nil)
+        descriptionLabel.linkDelegate = self
+        
+        // long press to copy the post text
+        let textLongPressGesture = UILongPressGestureRecognizer(target: self, action: #selector(self.onTextLongPress))
```
Comment (rileyhCode) [inline Mammoth/Screens/ProfileScreen/Views/ProfileHeader.swift:720]: This is new functionality, correct?

File: Mammoth/Views/Cells/UserCardCell/UserCardCell.swift [hunk 2]
```diff
@@ -73,18 +73,13 @@ final class UserCardCell: UITableViewCell {
...
         label.textColor = .custom.mediumContrast
-        label.enabledTypes = [.mention, .hashtag, .url, .email]
         label.numberOfLines = 2
-        label.mentionColor = .custom.highContrast
-        label.hashtagColor = .custom.highContrast
-        label.URLColor = .custom.highContrast
-        label.emailColor = .custom.highContrast
-        label.linkWeight = .semibold
-        label.urlMaximumLength = 30
+        label.textContainer.maximumNumberOfLines = 2
```
Comment (rileyhCode) [inline Mammoth/Views/Cells/UserCardCell/UserCardCell.swift:80]: Just confirming this change to set the max # of lines is expected.

**Review by rileyhCode — APPROVED** [review]
A couple notes for review, then feel free to merge.

File: Mammoth/Views/Cells/UserCardCell/UserCardCell.swift (hunk 2 above)
Comment (bnolens) [reply Mammoth/Views/Cells/UserCardCell/UserCardCell.swift:80]: Yes that's correct. We already set the number of lines (see L79) but with this new label we need to set it on the textContainer instead.

**Pull request merged**